pkg/resource/endpoint/custom_set_output.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Use this file if the Status/Spec of the CR needs to be modified after
+// create/describe/update operation
+
+package endpoint
+
+import (
+	"context"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// customCreateEndpointSetOutput sets the resource in TempOutofSync if endpoint is
+// in creating state. At this stage we know call to createEndpoint was successful.
+func (rm *resourceManager) customCreateEndpointSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.CreateEndpointOutput,
+	ko *svcapitypes.Endpoint,
+) (*svcapitypes.Endpoint, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.EndpointStatusCreating), ko)
+	return ko, nil
+}
+
+// customDescribeEndpointSetOutput sets the resource in TempOutofSync if endpoint is
+// being modified by AWS
+func (rm *resourceManager) customDescribeEndpointSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.DescribeEndpointOutput,
+	ko *svcapitypes.Endpoint,
+) (*svcapitypes.Endpoint, error) {
+	rm.customSetOutput(r, resp.EndpointStatus, ko)
+	return ko, nil
+}
+
+// customUpdateEndpointSetOutput sets the resource in TempOutofSync if endpoint is
+// being updated. At this stage we know call to updateEndpoint was successful.
+func (rm *resourceManager) customUpdateEndpointSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.UpdateEndpointOutput,
+	ko *svcapitypes.Endpoint,
+) (*svcapitypes.Endpoint, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.EndpointStatusUpdating), ko)
+	return ko, nil
+}
+
+// customSetOutput sets ConditionTypeResourceSynced condition to True or False
+// based on the endpoint status on AWS so the reconciler can determine if a
+// requeue is needed
+func (rm *resourceManager) customSetOutput(
+	r *resource,
+	endpointStatus *string,
+	ko *svcapitypes.Endpoint,
+) {
+	if endpointStatus != nil {
+		syncConditionStatus := corev1.ConditionUnknown
+		switch *endpointStatus {
+		case svcsdk.EndpointStatusInService, svcsdk.EndpointStatusFailed:
+			syncConditionStatus = corev1.ConditionTrue
+			break
+		default:
+			syncConditionStatus = corev1.ConditionFalse
+		}
```

Comment:
Guessing you expected there to be more conditions? Otherwise this could just be an `if-else`